manim/utils/hashing.py

```diff
@@ -41,8 +41,12 @@ def default(self, obj):
                 # NOTE : All module types objects are removed, because otherwise it throws ValueError: Circular reference detected if not. TODO
                 if isinstance(cvardict[i], ModuleType):
                     del cvardict[i]
+            try:
```

Comment:
@huguesdevimeux This was modified in #315 and actually breaks generating the documentation here: one video for `ArrowTip` is generated from a doctest, which is read as code entered from the command line. `inspect.getsource(obj)` throws an `OSError` for this thing (I think it tries to look at a file where the code is stored, which is not possible for commands entered on the prompt -- and thus fails).

These changes I made here were just a triage to test building the docs, I am not sure whether this is a good idea -- maybe you have an opinion? :-)

Reply:
In fact, this issue has has been pointed out previously in #348, an we fixed it by disabling scene cachiing when the code was passed through `-`
To be very frank, i'm not a big fan of this change as it will potentially create hash collisions when function's code won't be retrieved (like, instead of raising an issue, it will just set the code to an empty string, to the letter won't have an y influence on the hash).

But I guess it's the only simple solution, so it is good to me. Could you just add a comment explaining this ? 

PS: sorry for the late reply

Reply:
I agree, its not my favorite solution either; as I've said, this was more of a quick triage in order to make the docbuild process work again.

In principle, the code for generating the scene _is_ available -- after all, it is used in the `exec(...)` call! To avoid hash collision, I could hash it and pass the hash when initializing the scene... but I'd rather investigate this further in a follow-up issue.

I'll add a comment explaining what happens there (and why). Thanks for the feedback! :-)

Reply:
2d07edb4